app/src/test/java/com/lastaoutdoor/lasta/viewmodel/DiscoveryScreenViewModelTest.kt

```diff
@@ -1,70 +1,59 @@
-    // Given
-    val centerLocation = LatLng(46.519962, 6.633597)
-    val rad = 10000.0
-    val climbingNodes =
+  fun testInitialValues() {
+    assertEquals(viewModel.activities.value.isEmpty(), false) // Check initial activities
+    assertEquals(viewModel.screen.value, DiscoverDisplayType.LIST)
+    assertEquals(viewModel.range.value, 10000.0)
+    assertEquals(
+        viewModel.localities,
         listOf(
-            dummyNode(ActivityType.CLIMBING, "Climbing Node 1", centerLocation),
-            dummyNode(ActivityType.CLIMBING, "Climbing Node 2", centerLocation))
-    repository.addClimbingNode(climbingNodes[0])
-    repository.addClimbingNode(climbingNodes[1])
-
-    val method: Method =
-        DiscoveryScreenViewModel::class
-            .java
-            .getDeclaredMethod("fetchClimbingActivities", Double::class.java, LatLng::class.java)
-    method.isAccessible = true
-    method.invoke(viewModel, rad, centerLocation)
+            "Ecublens" to LatLng(46.519962, 6.633597),
+            "Geneva" to LatLng(46.2043907, 6.1431577),
+            "Payerne" to LatLng(46.834190, 6.928969),
+            "Matterhorn" to LatLng(45.980537, 7.641618)))
+  }
 
-    // Then
-    assertEquals(climbingNodes.size, viewModel.climbingActivities.value.size)
+  @Test
+  fun testPUpdatePermissions() {
+    viewModel.updatePermission(true)
+    assertEquals(viewModel.initialPosition, LatLng(46.519962, 6.633597))
+    viewModel.updatePermission(false)
+    assertEquals(viewModel.initialPosition, LatLng(46.519962, 6.633597))
   }
 
   @Test
-  fun `fetchClimbingActivities with no arguments should populate climbingActivities`() =
-      runBlocking {
-        // Given
-        val centerLocation = LatLng(46.519962, 6.633597)
-        val rad = 10000.0
-        val climbingNodes =
-            listOf(
-                dummyNode(ActivityType.CLIMBING, "Climbing Node 1", centerLocation),
-                dummyNode(ActivityType.CLIMBING, "Climbing Node 2", centerLocation))
-        repository.addClimbingNode(climbingNodes[0])
-        repository.addClimbingNode(climbingNodes[1])
+  fun testClearSelectedItinerary() {
+    viewModel.clearSelectedItinerary()
+    assertEquals(viewModel.selectedItinerary.value, null)
+  }
 
-        val method: Method =
-            DiscoveryScreenViewModel::class
-                .java
-                .getDeclaredMethod(
-                    "fetchClimbingActivities", Double::class.java, LatLng::class.java)
-        method.isAccessible = true
-        method.invoke(viewModel, rad, centerLocation)
+  @Test
+  fun testClearSelectedMarker() {
+    viewModel.clearSelectedMarker()
+    assertEquals(viewModel.selectedMarker.value, null)
+  }
 
-        // Then
-        assertEquals(climbingNodes.size, viewModel.climbingActivities.value.size)
-      }*/
+  @Test
+  fun testUpdateMarkers() {
+    viewModel.updateMarkers(LatLng(46.519962, 6.633597), 10000.0)
```

Comment:
You could have used Lausanne constant talked about earlier here!

Reply:
It would be nice a idea to create a constant for Ecublens indeed!